tests/unit_tests/grid/test_cell_properties.py

```diff
@@ -0,0 +1,281 @@
+    property_collection.write_hdf5_for_imported_list()
+    property_collection.create_xml_for_imported_list_and_add_parts_to_model()
+    if hasattr(grid, 'array_thickness'):
+        delattr(grid, 'array_thickness')
+
+    # Act
+    thickness = cp.thickness(grid, property_collection = property_collection)
+
+    # Assert
+    np.testing.assert_array_almost_equal(thickness, thickness_array)
+
+
+def test_thickness_multiple_property_collection(example_model_with_properties: Model):
+    # Arrange
+    grid = example_model_with_properties.grid()
+    extent = grid.extent_kji
+    property_collection = grid.property_collection
+    thickness_array_gross = np.random.random(extent)
+    property_collection.add_cached_array_to_imported_list(thickness_array_gross,
+                                                          'test data',
+                                                          'DZ',
+                                                          False,
+                                                          uom = grid.z_units(),
+                                                          property_kind = 'thickness',
+                                                          facet_type = 'netgross',
+                                                          indexable_element = 'cells',
+                                                          facet = 'gross')
+    thickness_array_net = np.random.random(extent) / 2
+    property_collection.add_cached_array_to_imported_list(thickness_array_net,
+                                                          'test data',
+                                                          'DZ',
+                                                          False,
+                                                          uom = grid.z_units(),
+                                                          property_kind = 'thickness',
+                                                          facet_type = 'netgross',
+                                                          indexable_element = 'cells',
+                                                          facet = 'net')
+    property_collection.write_hdf5_for_imported_list()
+    property_collection.create_xml_for_imported_list_and_add_parts_to_model()
+    if hasattr(grid, 'array_thickness'):
+        delattr(grid, 'array_thickness')
+
+    # Act
+    thickness = cp.thickness(grid, property_collection = property_collection)
+
+    # Assert
+    np.testing.assert_array_almost_equal(thickness, thickness_array_gross)
+
+
+def test_thickness_from_points(example_model_with_properties: Model):
+    # Arrange
+    grid = example_model_with_properties.grid()
+    if hasattr(grid, 'array_thickness'):
+        delattr(grid, 'array_thickness')
+    if hasattr(grid, 'property_collection'):
+        delattr(grid, 'property_collection')
+
+    # Act
+    thickness = cp.thickness(grid)
+
+    # Assert
+    np.testing.assert_array_almost_equal(thickness, 20.0)
```

Comment:
Surprised assert_array_almost_equal works comparing against a float?

Reply:
numpy has the concept of 'array like' which will take a scalar and treat it as a constant array when needed, if I've understood correctly, so I prefer the form Max has used here

Reply:
OK, in that case I retract my comments :-).